source/common/common/assert.h

```diff
@@ -3,21 +3,34 @@
 #include "common/common/logger.h"
 
 namespace Envoy {
+
 /**
  * assert macro that uses our builtin logging which gives us thread ID and can log to various
  * sinks.
+ *
+ * The old style release assert was of the form RELEASE_ASSERT(foo == bar);
+ * where it would log stack traces and the failed conditional and crash if the
+ * condition is not met.  The are many legacy RELEASE_ASSERTS in Envoy which
+ * were converted to RELEASE_ASSERT(foo == bar, "");
+ *
+ * The new style of release assert is of the form
+ * RELEASE_ASSERT(foo == bar, "reason foo should actually be bar");
+ * new uses of RELEASE_ASSERT are strongly encouraged to supply a verbose
```

Comment:
I think this comment applies to one of your previous iterations. This is a required param of the macro, right?

Reply:
I think this is correct, explaining the old macro and legacy ""s.  I've made wording stronger for "include details already" :-)